src/transforms/transform-deployment.ts

```diff
@@ -48,30 +53,90 @@ const getLastSuccessfulDeployCommitSha = async (
+			logger.info("Couldn't find last success deployment from dynamodb");
+			statsd.increment(metricDeploymentPersistent.miss, { missedType: "not-found", ...tags }, info);
+			return undefined;
+		} else if (!lastSuccessful.commitSha) {
+			logger.warn("Missing commit sha from deployment");
+			statsd.increment(metricDeploymentPersistent.miss, { missedType: "sha-empty", ...tags }, info);
+			return undefined;
+		} else {
+			logger.info("Found last success deployment info");
+			statsd.increment(metricDeploymentPersistent.hit, tags, info);
+			return lastSuccessful.commitSha;
+		}
+
+	} catch (e) {
+		statsd.increment(metricDeploymentPersistent.failed, { failType: "lookup", ...tags }, info);
+		logger.error({ err: e }, "Error look up deployment information from dynamodb");
+		throw e;
+	}
+};
+
 const getCommitsSinceLastSuccessfulDeployment = async (
+	jiraHost: string,
+	type: "backfill" | "webhook",
 	owner: string,
+	repoId: number,
 	repoName: string,
 	currentDeploySha: string,
 	currentDeployId: number,
 	currentDeployEnv: string,
+	currentDeployDate: string,
 	githubInstallationClient: GitHubInstallationClient,
 	logger: Logger
 ): Promise<CommitSummary[] | undefined> => {
 
-	// Grab the last 10 deployments for this repo
-	const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> | AxiosResponse<Octokit.ReposListDeploymentsResponse> =
-		await githubInstallationClient.listDeployments(owner, repoName, currentDeployEnv, 10);
-
-	// Filter per current environment and exclude itself
-	const filteredDeployments = deployments.data
-		.filter(deployment => deployment.id !== currentDeployId);
+	let lastSuccessfullyDeployedCommit: string | undefined = undefined;
 
-	// If this is the very first successful deployment ever, return nothing because we won't have any commit sha to compare with the current one.
-	if (!filteredDeployments.length) {
-		return undefined;
+	if (type === "webhook" && await booleanFlag(BooleanFlags.USE_DYNAMODB_FOR_DEPLOYMENT_WEBHOOK, jiraHost)) {
+		lastSuccessfullyDeployedCommit = await getLastSuccessDeploymentShaFromDyanmoDB(type, jiraHost, repoId, currentDeployEnv, currentDeployDate, githubInstallationClient, logger);
 	}
 
-	const lastSuccessfullyDeployedCommit = await getLastSuccessfulDeployCommitSha(owner, repoName, githubInstallationClient, filteredDeployments, logger);
+	if (!lastSuccessfullyDeployedCommit) {
+		// Grab the last 10 deployments for this repo
+		const deployments: Octokit.Response<Octokit.ReposListDeploymentsResponse> | AxiosResponse<Octokit.ReposListDeploymentsResponse> =
+			await githubInstallationClient.listDeployments(owner, repoName, currentDeployEnv, 10);
+
+		// Filter per current environment and exclude itself
+		const filteredDeployments = deployments.data
+			.filter(deployment => deployment.id !== currentDeployId);
+
+		// If this is the very first successful deployment ever, return nothing because we won't have any commit sha to compare with the current one.
```

Comment:
amen!

Reply:
Just to understand that correctly, if item is not found, we are falling back to the old logic, right?